tags/Adcrowd/Adcrowd.tpl

```diff
@@ -0,0 +1,117 @@
+    "displayName": "",
+    "id": "brand_dummy"
+  },
+  "containerContexts": [
+    "WEB"
+  ]
+}
+
+
+___TEMPLATE_PARAMETERS___
+
+[
+  {
+    "help": "This is the unique ID in the pixel: //pixel.adcrowd.com/smartpixel/{{PIXELID}}.js",
+    "valueValidators": [
+      {
+        "errorMessage": "The Pixel ID is required for Adcrowd to work",
+        "type": "NON_EMPTY"
+      }
+    ],
+    "displayName": "Pixel ID",
+    "simpleValueType": true,
+    "name": "pixelId",
+    "type": "TEXT"
+  },
+  {
+    "help": "Populate with the product ID on the product page. Needs to correspond with the product ID in the product feed. Can only return single product ID.",
+    "displayName": "Product ID",
+    "simpleValueType": true,
+    "name": "productId",
+    "type": "TEXT"
+  },
+  {
+    "displayName": "Order ID",
+    "simpleValueType": true,
+    "name": "orderId",
+    "type": "TEXT"
+  },
+  {
+    "displayName": "Order Value",
+    "simpleValueType": true,
+    "name": "orderValue",
+    "type": "TEXT"
+  }
+]
+
+
+___WEB_PERMISSIONS___
+
+[
+  {
+    "instance": {
+      "key": {
+        "publicId": "inject_script",
+        "versionId": "1"
+      },
+      "param": [
+        {
+          "key": "urls",
+          "value": {
+            "type": 2,
+            "listItem": [
+              {
+                "type": 1,
+                "string": "https://pixel.adcrowd.com/"
+              }
+            ]
+          }
+        }
+      ]
+    },
+    "clientAnnotations": {
+      "isEditedByUser": true
+    },
+    "isRequired": true
+  }
+]
+
+
+___SANDBOXED_JS_FOR_WEB_TEMPLATE___
+
+// Enter your template code here.
+//var log = require('logToConsole');
+const injectScript = require('injectScript');
+
+var _ac = {};
+if(data.orderValue !== undefined){
+	_ac.value = data.orderValue;
+}
+if(data.productId !== undefined){
+	_ac.product = data.productId;
+}
+if(data.orderId !== undefined){
+	_ac.order_id = data.orderId;
+}
+
```

Comment:
Would probably be more legible if you refactor it to:

 ```
const _ac = {
  value: data.orderValue,
  product: data.productId,
  order_id: data.orderId
};
```

This should work fine unless there's some specific checking to see if fields contain undefined values.

Reply:
That's correct. In some cases, fields may be undefined.

Reply:
Yeah but setting them in the object even if they're undefined should be OK as the JS library should be structured in a way to automatically exclude undefined items (e.g. how Facebook, GA works). Naturally, if this is not the case and `undefined` is significant, then the original way of conditionally populating the object is better :)